spyder/workers/updates.py

```diff
@@ -5,37 +5,49 @@
 # (see spyder/__init__.py for details)
 
 # Standard library imports
-import json
 import logging
 import os
 import os.path as osp
-import ssl
 import tempfile
 import traceback
-from urllib.request import urlopen, urlretrieve
-from urllib.error import URLError, HTTPError
 
 # Third party imports
 from qtpy.QtCore import QObject, Signal
+import requests
+from requests.adapters import ConnectionError, SSLError
 
 # Local imports
 from spyder import __version__
 from spyder.config.base import (
     _, is_stable_version, is_pynsist, running_in_mac_app)
-from spyder.py3compat import is_text_string
 from spyder.config.utils import is_anaconda
 from spyder.utils.conda import get_spyder_conda_channel
 from spyder.utils.programs import check_version, is_module_installed
 
 # Logger setup
 logger = logging.getLogger(__name__)
 
+SSL_ERROR_MSG = _(
+    'SSL certificate verification failed while checking for Spyder updates.'
+    '<br><br>Please contact your network administrator for assistance.'
+)
+
+CONNECT_ERROR_MSG = _(
+    'Unable to connect to the internet while checking for Spyder updates.'
```

Comment:
This should say something more like "Unable to connect to the Spyder update service." since this could be something specific to that site or due to a firewall, proxy, etc. issue rather than just internet connectivity